From 9c73eb7c3562ac84f88dc70d2172ab1f1428448a Mon Sep 17 00:00:00 2001
From: Ayman Grais <ayman.grais@morsemicro.com>
Date: Wed, 29 Apr 2026 04:25:18 +0000
Subject: [PATCH] SW-19927 (rel_mm8108_2_0) Disable A-MSDU for Thin LMAC (pull
 request #2109)

- Thin-LMAC does not support A-MSDU (at least for now)
- It can still be re-enabled via setting enable_amsdu_override modparam

Cherry picked from 072ab51
---
 mac.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mac.c b/mac.c
index cba1197d7..3464d3e0e 100644
--- a/mac.c
+++ b/mac.c
@@ -991,6 +991,11 @@ bool morse_mac_is_amsdu_enabled(struct morse *mors)
 			return enable_amsdu_override;
 		}
 
+		if (is_thin_lmac_mode() && mors->cfg->enable_amsdu_support) {
+			MORSE_INFO(mors, "%s: AMSDU disabled in Thin LMAC mode\n", __func__);
+			return false;
+		}
+
 		MORSE_INFO(mors, "%s: AMSDU support: %s\n",
 			__func__,
 			mors->cfg->enable_amsdu_support ? "true" : "false");
-- 
2.34.1

